waterdata/filters.py

```diff
@@ -40,37 +42,45 @@ def use_correct_indefinite_article(some_noun):
 
 
 @app.template_filter('data_start_year')
-def data_start_year(series):
+def data_start_year(parameter_group_series):
     """
     Determine the earliest year that was collected for a
     list of parameters.
 
-    :param list series: list of data series
-    :return: earliest year of data collection, list of distinct things being measured
-    :rtype: str
+    :param dict parameter_group_series: dictionary of parameter series grouped by parameter group
+    :return: earliest year of data collection if available
+    :rtype: str or None
 
     """
+    series = chain.from_iterable([x['parameters'] for x in parameter_group_series.values()])
     all_start_dates = [s['start_date'] for s in series]
     try:
         start_year = str(min(all_start_dates).year)
     except ValueError:
-        start_year = 'unknown'
+        start_year = None
     return start_year
 
 
 @app.template_filter('readable_param_list')
-def readable_param_list(series):
+def readable_param_list(parameter_group_series):
     """
     Generate text for a list of parameters in the description meta tag.
 
-    :param list series: list of data series
-    :return: human readable string for the parameters in the series
-    :rtype: str
+    :param dict parameter_group_series: dictionary of parameter series grouped by parameter group
+    :return: human readable string for the parameters in the series if available
+    :rtype: str or None
 
     """
     # need to use a set -- there might be multiple variants for a measurements
     # (e.g. "nitrite, water as N" vs "nitrite, water as NO3-")
-    short_names = set([s['parameter_name'].split(',')[0].upper() for s in series])
+    series = chain.from_iterable([x['parameters'] for x in parameter_group_series.values()])
+    short_names = set(
+        [
+            s['parameter_name'].split(',')[0].upper() for s in series if 'Unit Values' in s['data_types'] and
+            (datetime.datetime.now().date() - datetime.timedelta(days=8) <=
+             s['end_date'].date() <= datetime.datetime.now().date() + datetime.timedelta(days=1))
```

Comment:
Mixed feelings about the date filtering here. There are timedeltas to handle instances if code is being executed around the site is in a different timezone.

Ideally, the comparison here would be done in UTC, but I couldn't think of a way to get a site's UTC offset without adding at least another service call. I wagered it wasn't worth it, since this will probably be okay most of the time.

Reply:
I think this is fine for purposes of this ticket.